src/constructs/autoscaling/asg.ts

```diff
@@ -107,7 +109,7 @@ export class GuAutoScalingGroup extends AutoScalingGroup {
       userData,
       // Do not use the default AWS security group which allows egress on any port.
       // Favour HTTPS only egress rules by default.
-      securityGroup: GuHttpsEgressSecurityGroup.forVpc(scope, props),
+      securityGroup: GuHttpsEgressSecurityGroup.forVpc(scope, { app: props.app, vpc: props.vpc }),
```

Comment:
As [commented on](https://github.com/guardian/cdk/pull/400/files#r609667036) in #400:

> Passing in props meant existingLogicalId was copied to the GuHttpsEgressSecurityGroup too. When it came time to synth the stack, it failed because two resources were being created with the same logicalId (the GuHttpsEgressSecurityGroup and the GuAutoScalingGroup).
> 
> This took the best part of two days to diagnose 😨 !